include/odp/api/spec/pool_ext.h

```diff
@@ -0,0 +1,318 @@
+ * @param param    Pool parameters
+ *
+ * @return Pool handle on success
+ * @retval ODP_POOL_INVALID on failure
+ */
+odp_pool_t odp_pool_ext_create(const char *name, const odp_pool_ext_param_t *param);
+
+/**
+ * Populate external memory pool with buffer memory
+ *
+ * Populate can be called multiple times to add memory buffers into the pool. Application must
+ * populate the pool with the exact number of buffers specified in pool parameters. The pool is
+ * ready to be used for allocations only after all populate calls have returned successfully.
+ *
+ * Pool type defines memory buffer layout and where the buffer pointer (buf[N]) points
+ * in the layout. Pool capabilities specify requirements for buffer size, layout and
+ * pointer alignment.
+ *
+ * For packet pools, packet buffer layout is shown under. The packet headroom (odp_packet_head())
+ * starts immediately after the application header. For a segmented packet, each segment has this
+ * same layout.
+ *
+ * @code{.unparsed}
+ *
+ *         buf[N] --->  +-------------------------------+    --                     --
+ *                      |                               |     |                      |
+ *                      | ODP header (optional)         |      > odp_header_size     |
+ *                      |                               |     |                      |
+ *                      +-------------------------------+    --                      |
+ *                      |                               |     |                      |
+ *                      | Application header (optional) |      > app_header_size     |
+ *                      |                               |     |                       > buf_size
+ *                      +-------------------------------+    --                      |
+ * odp_packet_head()--> |                               |                            |
+ *                      | Packet data                   |                            |
+ *                      |  (headroom, data, tailroom)   |                            |
+ *                      |                               |                            |
+ *                      |                               |                            |
+ *                      +-------------------------------+    --                      |
+ *                      |                               |     |                      |
+ *                      | ODP trailer (optional)        |      > odp_trailer_size    |
+ *                      |                               |     |                      |
+ *                      +-------------------------------+    --                     --
+ *
+ * @endcode
+ *
+ * @param pool    External memory pool
+ * @param buf     Buffer pointers to be populated into the pool
+ * @param num     Number of buffer pointers
+ * @param flags   Flags reserved for future use.
+ *                TODO: will additional information be needed per buffer? E.g.
```

Comment:
Please add one DONE flag. To denote the completion of populate.
i.e after DONE odp_packet_alloc() etc can be called.

Reply:
done in v8